test/unit/core/DatetimeClasses.js

```diff
@@ -502,4 +502,253 @@ describe('DateTimeClassesTest', function () {
+                (() => new LocalTime(1, 1, 1, 1)).should.not.throw();
+
+                (() => new LocalTime(1, 58, 1, 1)).should.not.throw();
+                (() => new LocalTime(1, 59, 1, 1)).should.not.throw();
+                (() => new LocalTime(1, 60, 1, 1)).should.throw(RangeError, 'Minute');
+            });
+
+            it('second', function() {
+                // [0-59]
+                (() => new LocalTime(1, 1, -1, 1)).should.throw(RangeError, 'Second');
+                (() => new LocalTime(1, 1, 0, 1)).should.not.throw();
+                (() => new LocalTime(1, 1, 1, 1)).should.not.throw();
+
+                (() => new LocalTime(1, 1, 58, 1)).should.not.throw();
+                (() => new LocalTime(1, 1, 59, 1)).should.not.throw();
+                (() => new LocalTime(1, 1, 60, 1)).should.throw(RangeError, 'Second');
+            });
+
+            it('nanosecond', function() {
+                // [0-999999999]
+                (() => new LocalTime(1, 1, 1, -1)).should.throw(RangeError, 'Nano');
+                (() => new LocalTime(1, 1, 1, 0)).should.not.throw();
+                (() => new LocalTime(1, 1, 1, 1)).should.not.throw();
+
+                (() => new LocalTime(1, 1, 1, 999999998)).should.not.throw();
+                (() => new LocalTime(1, 1, 1, 999999999)).should.not.throw();
+                (() => new LocalTime(1, 1, 1, 999999999 + 1)).should.throw(RangeError, 'Nano');
+            });
+        });
+        describe('LocalDateTest', function () {
+            it('year', function() {
+                // [-999999999-999999999]
+                (() => new LocalDate(-(999999999+1), 1, 1)).should.throw(RangeError, 'Year');
+                (() => new LocalDate(-999999999, 1, 1)).should.not.throw();
+                (() => new LocalDate(-999999998, 1, 1)).should.not.throw();
+
+                (() => new LocalDate(999999998, 1, 1)).should.not.throw();
+                (() => new LocalDate(999999999, 1, 1)).should.not.throw();
+                (() => new LocalDate(999999999 + 1, 1, 1)).should.throw(RangeError, 'Year');
+            });
+
+            it('month', function() {
+                // [1-12]
+                (() => new LocalDate(1, 0, 1)).should.throw(RangeError, 'Month');
+                (() => new LocalDate(1, 1, 1)).should.not.throw();
+                (() => new LocalDate(1, 2, 1)).should.not.throw();
+
+                (() => new LocalDate(1, 11, 1)).should.not.throw();
+                (() => new LocalDate(1, 12, 1)).should.not.throw();
+                (() => new LocalDate(1, 13, 1)).should.throw(RangeError, 'Month');
+            });
+
+            it('day', function() {
+                // [1-31]
+                (() => new LocalDate(1, 1, 0)).should.throw(RangeError, 'Date');
+                (() => new LocalDate(1, 1, 1)).should.not.throw();
+                (() => new LocalDate(1, 1, 2)).should.not.throw();
+
+                (() => new LocalDate(1, 1, 30)).should.not.throw();
```

Comment:
How about `LocalDate(1, 2, 29)`? Or `LocalDate(1, 4, 31)`?

Reply:
Such cases are tested in other tests in the same file. However I did not test all 12 months